src/libstore/build/local-derivation-goal.cc

```diff
@@ -1442,8 +1442,17 @@ void LocalDerivationGoal::startDaemon()
 
 void LocalDerivationGoal::stopDaemon()
 {
-    if (daemonSocket && shutdown(daemonSocket.get(), SHUT_RDWR) == -1)
+    if (daemonSocket && shutdown(daemonSocket.get(), SHUT_RDWR) == -1) {
+        // on macOS if we are just _waiting_ (accept) for a connection
+        // the socced is not considered connected. Hence shutdown will
+        // complain that it's not connected. Thus simply close it. Which
+        // will also exit the accept call.
+        if(errno != ENOTCONN) {
+            close(daemonSocket.get());
```

Comment:
```suggestion
            close(daemonSocket.get());
            daemonSocket.release();
```

The sooner the better, I think. cc @edolstra